### PR TITLE
Use binary search when searching for lines indexes in LetVarWhiteSpacesRule

### PR DESCRIPTION
**Pull request by PaulTaykalo**
A bit faster searching for an line that starts specified offset for `LetVarWhitespaceRule`
While it doesn't seems a big deal, on big files, this gives a big boost in performance

### CODE REVIEW

**Comment by PaulTaykalo** [comment]
Some samples on the local machine, `LetVarWhitespaceRule` is used eveywhere so it affects a lot of repositories
```
Message: Linting Alamofire with this PR took 5.81s vs 8.22s on master (29% faster)
Master best 6.86: avg :8.22, samples: [6.855083, 8.198644, 8.151273, 7.447268, 7.551027, 9.199949, 9.650616, 9.20073, 7.850128, 8.05038]
Branch best 4.89: avg :5.81, samples: [8.299909, 6.345781, 5.850497, 5.398514, 5.549249, 5.501168, 5.70256, 5.356872, 4.88762, 5.201307]
Message: Linting Firefox with this PR took 17.33s vs 21.63s on master (19% faster)
Master best 19.13: avg :21.63, samples: [22.40823, 21.344212, 20.250126, 19.814322, 19.134326, 19.197361, 21.053172, 26.696117, 23.052895, 23.302886]
Branch best 14.65: avg :17.33, samples: [16.636716, 15.96878, 16.249455, 14.649814, 15.562665, 20.536516, 16.01, 20.790482, 17.903025, 18.998291]
Message: Linting Kickstarter with this PR took 43.0s vs 49.67s on master (13% faster)
Master best 40.31: avg :49.67, samples: [45.606528, 46.744854, 48.206579, 61.845327, 65.098365, 45.592601, 40.307346, 49.193308, 51.314768, 42.832952]
Branch best 38.13: avg :43.0, samples: [50.362238, 46.996996, 38.590681, 38.205356, 40.610003, 43.691837, 50.558141, 38.132902, 44.149775, 38.700872]
Message: Linting WordPress with this PR took 50.29s vs 47.16s on master (6% slower)
Master best 41.76: avg :47.16, samples: [54.409536, 50.836631, 49.805967, 56.145498, 47.495008, 41.912286, 44.585946, 42.010824, 42.639966, 41.761133]
Branch best 39.19: avg :50.29, samples: [78.563618, 62.637871, 40.457022, 39.345795, 44.703578, 39.189002, 51.5194, 53.129123, 42.241193, 51.119719]
```

File: Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift [hunk 1]
```diff
@@ -225,8 +225,12 @@ private extension SwiftDeclarationKind {
 }
 
 private extension File {
-    // Zero-based line number for the given a byte offset
     func line(byteOffset: Int, startFrom: Int = 0) -> Int {
+        return lineFast(byteOffset: byteOffset, startFrom: startFrom)
+    }
+
+    // Zero-based line number for the given a byte offset
+    func lineSlow(byteOffset: Int, startFrom: Int = 0) -> Int {
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift]: it doesn't look like this is used, can we remove it?

**Comment by PaulTaykalo** [comment]
@marcelofabri Done

**Comment by PaulTaykalo** [comment]
@marcelofabri Is there anything I can help with to merge this one?

File: Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift [hunk 2]
```diff
@@ -225,15 +225,43 @@ private extension SwiftDeclarationKind {
 }
 
 private extension File {
-    // Zero-based line number for the given a byte offset
     func line(byteOffset: Int, startFrom: Int = 0) -> Int {
-        for index in startFrom..<lines.count {
+        return lineFast(byteOffset: byteOffset, startFrom: startFrom)
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift]: what if we use `contents.bridge().lineAndCharacter(forByteOffset: byteOffset)?.line).map { $0 - 1 }`? 

This does a binary search as well.

Comment (PaulTaykalo) [reply Source/SwiftLintFramework/Rules/Style/LetVarWhitespaceRule.swift]: It seems to be working as well.

**Comment by jpsim** [comment]
🎉 

Nice win!

**Pull request merged**